src/H5Olinfo.c

```diff
@@ -140,6 +140,9 @@ H5O__linfo_decode(H5F_t *f, H5O_t H5_ATTR_UNUSED *open_oh, unsigned H5_ATTR_UNUS
         if (H5_IS_BUFFER_OVERFLOW(p, 8, p_end))
             HGOTO_ERROR(H5E_OHDR, H5E_OVERFLOW, NULL, "ran off end of input buffer while decoding");
         INT64DECODE(p, linfo->max_corder);
+        if (linfo->max_corder < 0)
+            HGOTO_ERROR(H5E_OHDR, H5E_BADVALUE, NULL, "invalid max creation order value for message: %lld",
+                        (long long)linfo->max_corder);
```

Comment:
Can be printed with `message: %" PRId64` here rather than casting

Reply:
Fixed